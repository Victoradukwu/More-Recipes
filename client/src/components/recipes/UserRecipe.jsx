import PropTypes from 'prop-types';
import React from 'react';
import { Link } from 'react-router-dom';

const UserRecipe = ({ userRecipe, deleteRecipe }) => (
  <tr>
    <td>
      <Link to={`/recipe/${userRecipe.id}`}>{userRecipe.recipeName}</Link>
    </td>
    <td>{userRecipe.upvote}</td>
    <td>{userRecipe.downvote}</td>
    <td>{userRecipe.favorites}</td>
    <td>{userRecipe.views}</td>
    <td>
      <button
        className="btn btn-danger"
        onClick={() =>
        deleteRecipe(userRecipe.id)}
      >Delete
      </button>
    </td>

  </tr>
);

UserRecipe.propTypes = {
  userRecipe: PropTypes.object.isRequired,
  deleteRecipe: PropTypes.func.isRequired,
};
export default UserRecipe;
