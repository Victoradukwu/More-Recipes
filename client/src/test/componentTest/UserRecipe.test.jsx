import React from 'react';
import { shallow } from 'enzyme';
import UserRecipe from '../../components/recipes/UserRecipe';

const props = {
  userRecipe: [{ id: 1 }],
  deleteRecipe: jest.fn(),
};

describe('UserRecipesPage', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<UserRecipe {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('simulates click', () => {
    const wrapper = shallow(<UserRecipe {...props} />);
    const btn = wrapper.find('button.btn.btn-danger');
    btn.simulate('click');
    expect(props.deleteRecipe).toHaveBeenCalled();
  });
});
