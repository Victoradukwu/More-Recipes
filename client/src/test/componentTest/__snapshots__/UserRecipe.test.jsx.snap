// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserRecipesPage matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserRecipe
    deleteRecipe={[MockFunction]}
    userRecipe={
      Array [
        Object {
          "id": 1,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <td>
          <Link
            className="recipeName"
            replace={false}
            to="/recipe/undefined"
          />
        </td>,
        <td />,
        <td />,
        <td />,
        <td />,
        <td>
          <button
            className="btn btn-danger"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>,
      ],
      "id": "userRecipe",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            className="recipeName"
            replace={false}
            to="/recipe/undefined"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
            "className": "recipeName",
            "replace": false,
            "to": "/recipe/undefined",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="btn btn-danger"
            onClick={[Function]}
          >
            Delete
          </button>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Delete",
            "className": "btn btn-danger",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Delete",
          "type": "button",
        },
        "type": "td",
      },
    ],
    "type": "tr",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <td>
            <Link
              className="recipeName"
              replace={false}
              to="/recipe/undefined"
            />
          </td>,
          <td />,
          <td />,
          <td />,
          <td />,
          <td>
            <button
              className="btn btn-danger"
              onClick={[Function]}
            >
              Delete
            </button>
          </td>,
        ],
        "id": "userRecipe",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="recipeName"
              replace={false}
              to="/recipe/undefined"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": undefined,
              "className": "recipeName",
              "replace": false,
              "to": "/recipe/undefined",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn btn-danger"
              onClick={[Function]}
            >
              Delete
            </button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Delete",
              "className": "btn btn-danger",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Delete",
            "type": "button",
          },
          "type": "td",
        },
      ],
      "type": "tr",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
